apis/apiextensions/v1/composition_environment.go

```diff
@@ -71,12 +79,40 @@ type EnvironmentSource struct {
 type EnvironmentSourceReference struct {
 	// The name of the object.
 	Name string `json:"name"`
+
+	// Policies for referencing.
+	// +optional
+	Policy *Policy `json:"policy,omitempty"`
 }
 
 // An EnvironmentSourceSelector selects an EnvironmentConfig via labels.
 type EnvironmentSourceSelector struct {
 	// MatchLabels ensures an object with matching labels is selected.
 	MatchLabels []EnvironmentSourceSelectorLabelMatcher `json:"matchLabels,omitempty"`
+
+	// Policies for referencing.
+	// +optional
+	Policy *Policy `json:"policy,omitempty"`
```

Comment:
The Policy is attached to the Source/Selector?  So different sources/selectors can have different Policies? Does any one source/selector having the policy attached trigger the entire environment to be recreated?  If so why not have the Policy at the Composition.environment level?

Reply:
makes sense, will adjust it